kmip/pie/factory.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2015 The Johns Hopkins University/Applied Physics Laboratory
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from kmip.core import attributes
+from kmip.core import misc
+from kmip.core import objects as cobjects
+from kmip.core import secrets
+
+from kmip.pie import objects as pobjects
+
+
+class ObjectFactory:
+    """
+    A factory to convert between the Pie and core object hierarchies.
+    """
+
+    def __init__(self):
+        """
+        Construct an ObjectFactory.
+        """
+        pass
+
+    def convert(self, obj):
+        """
+        Convert a Pie object into a core secret object and vice versa.
+
+        Args:
+            obj (various): A Pie or core secret object to convert into the
+                opposite object space. Required.
+
+        Raises:
+            TypeError: if the object type is unrecognized or unsupported.
+        """
+        if isinstance(obj, pobjects.SymmetricKey):
+            return self._build_core_key(obj, secrets.SymmetricKey)
+        elif isinstance(obj, secrets.SymmetricKey):
+            return self._build_pie_key(obj, pobjects.SymmetricKey)
+        elif isinstance(obj, pobjects.PublicKey):
+            return self._build_core_key(obj, secrets.PublicKey)
+        elif isinstance(obj, secrets.PublicKey):
+            return self._build_pie_key(obj, pobjects.PublicKey)
+        elif isinstance(obj, pobjects.PrivateKey):
+            return self._build_core_key(obj, secrets.PrivateKey)
+        elif isinstance(obj, secrets.PrivateKey):
+            return self._build_pie_key(obj, pobjects.PrivateKey)
+        else:
+            raise TypeError("object type unsupported and cannot be converted")
```

Comment:
We could do that. I do prefer the current approach, since the interface to the factory is so simple, but if you think this is compelling we can change it. What makes you uncomfortable about the switch?